compiler-plugin/src/main/kotlin/com/google/devtools/ksp/symbol/impl/synthetic/KSConstructorSyntheticImpl.kt

```diff
@@ -68,7 +73,14 @@ class KSConstructorSyntheticImpl(val ksClassDeclaration: KSClassDeclaration) : K
         ksClassDeclaration.location
     }
 
-    override val modifiers: Set<Modifier> = emptySet()
+    override val modifiers: Set<Modifier> by lazy {
+        // add public if parent class is public
+        if (ksClassDeclaration.isPublic()) {
+            setOf(Modifier.FINAL, Modifier.PUBLIC)
```

Comment:
This need some attention. If we set a class to be private, other classes in same file can still access its constructor, but if we set its constructor to be private, then other classes in same file won't be able to access its constructor. 

i.e. constructor's `private` modifier does not work as its corresponding class'

Reply:
i'm not setting it to private here though, only setting it to public if class is public and if not, just not adding anything.

Reply:
Yeah, this actually still works because kotlin treat default to be public. The point is a private class will have a different modifier behavior than its public class counterpart, while in fact both are public. A minor comment though.

Reply:
this part definitely needs a followup (recalling the delegation to descriptors discussion we head earlier)

Is there any changes I should make here or is it good enough for now?

Reply:
Merged.